controller/comparator.go

```diff
@@ -105,33 +115,42 @@ func (ks *KsonnetAppComparator) CompareAppState(
 		resState := v1alpha1.ResourceState{
 			ChildLiveResources: make([]v1alpha1.ResourceNode, 0),
 		}
-		targetObjBytes, err := json.Marshal(targetObjs[i].Object)
-		if err != nil {
-			return nil, err
+		diffResult := diffResults.Diffs[i]
+		if diffResult.Modified {
+			resState.Status = v1alpha1.ComparisonStatusOutOfSync
+		} else {
+			resState.Status = v1alpha1.ComparisonStatusSynced
 		}
-		resState.TargetState = string(targetObjBytes)
+
+		if targetObjs[i] == nil {
+			resState.TargetState = "null"
+			resState.Status = v1alpha1.ComparisonStatusOutOfSync
```

Comment:
I have no problems with the logic, but it takes some time to understand it. Can we add comments about each scenario and why we are considering it out of sync? For example:

1) object is in target, but does not exist in live
2) object is in target and live, but differ
3) object is not defined in target, but exists in the live state

Reply:
done